### ISSUE TITLE
Gateway restart never completes while /ix1 is wedged — Symphonym resolution probes symlinks into it

### ISSUE THREAD

**Issue by docuracy** [WorldHistoricalGazetteer/place#242]
## Summary

**A gateway restart while `/ix1` is unreachable never completes.** Both uvicorn workers enter uninterruptible `D` state, nothing ever binds 9200, and the process must be killed and reconfigured by hand. Observed in production on 5 September 2026: ~13 minutes of total outage, ended only by pinning `SYMPHONYM_MODEL_DIR` to a `/vast` copy.

This is the same defect class as place#241, which was about the *query* path. This one is in the *startup* path, and it is worse in one specific way: place#241 degraded individual requests, whereas this takes the whole service down and blocks its own recovery.

## Mechanism

`gateway/symphonym.py::_resolve_model_dir` probes the repo's `hf/` directory:

```python
# gateway/symphonym.py:47-50
if (repo_hf / "config.json").exists():
    has_weights = (repo_hf / "model.safetensors").exists() or (repo_hf / "final_model.pt").exists()
    has_vocab = (repo_hf / "vocab" / "char_vocab.json").exists()
```

On the deployed tree, two of those entries are **symlinks into `/ix1`**:

```
final_model.pt -> /ix1/ishi/models/phonetic/checkpoints/v7/final_model.pt
vocab          -> /ix1/ishi/models/phonetic/data/v7/vocab
```

`Path.exists()` follows a symlink, so each probe `stat()`s a hard-mounted NFS path. A hard mount that wedges does not raise `ENOENT` or `EIO` — it blocks in uninterruptible sleep, indefinitely.

Step 3 (`gateway/symphonym.py:56-63`) then probes `IX1_BASE` directly and has the same exposure, as does the eventual `SymphonymModel(model_dir=...)` load, which reads a 99 MB `final_model.pt` across the same mount.

**The symlinks are self-inflicted.** `_resolve_model_dir` step 3 creates them itself (`symlink_to`, lines 65-78), so a gateway that once resolved the CRC layout permanently rewires its own step 2 to traverse `/ix1`. They are dated 5 Jun 2026 on the live VM.

## Why the existing guard does not catch it

`gateway/app.py:115-121`:

```python
# Pre-warm the Symphonym model (lazy import — won't crash if unavailable)
try:
    from . import symphonym
    symphonym.get_model()
except Exception as e:
    logger.warning(f"Symphonym model not available: {e} — /api/search/phonetic will fail")
```

The comment states the intent exactly — *won't crash if unavailable* — and `except Exception` delivers it for every failure that **raises**. A hang raises nothing, so the handler never runs. Same shape, same wrong assumption, as the `hard_link_expansion` guard in place#241 and the `a-check-that-cannot-fail` class in `developer/postmortem-ingestion-faults.md`.

## Why it went unnoticed until now

A long-running gateway holds the model in RAM, so it rides out an `/ix1` outage with no symptom at all. **The restart is what kills it** — the process discards everything it acquired when the environment was healthy and re-acquires from the environment as it now is. Any restart during any `/ix1` incident would have done this, with or without place#241's deploy.

## Evidence

```
2026-09-05 08:16:08  Gateway starting — ES: http://localhost:9201
2026-09-05 08:22:24  Loading Symphonym model from /vast/ishi/elastic/hf...
                     (no further output; never listened)

PID     STAT  ELAPSED  CMD
2163075 Dl    06:18    ...spawn_main(tracker_fd=7, pipe_handle=9)
2163076 Dl    06:18    ...spawn_main(tracker_fd=7, pipe_handle=13)

$ ss -ltn | grep 9200      # nothing
```

## Current mitigation (in place, not a fix)

`/vast/ishi/elastic/.env`:

```
SYMPHONYM_MODEL_DIR="/vast/ishi/models/phonetic/symphonym-v7-hf"
```

That directory is a self-contained copy fetched from the published `docuracy/symphonym-v7`; its `vocab/*.json` are sha256-identical to `/vast/ishi/models/phonetic/data/v7/vocab/`. The gateway now loads in 1.8 s with no `/ix1` touch.

⚠️ This only routes *around* the landmine. The `hf/` symlinks are still there, and step 1 falls through to step 2 whenever `SYMPHONYM_MODEL_DIR` is unset, wrong, or dropped — at which point the next restart hangs again.

## Proposed fix

1. Put every filesystem probe and the model load itself behind `gateway.bounded_io.IoGuard` (added for place#241), so an unresponsive candidate is *skipped* within seconds rather than blocking. A skipped candidate must not abort resolution — the next one is tried.
2. Prefer `IX3_BASE` over `IX1_BASE` in the CRC-layout branch, so the serving path defaults to flash.
3. Plant the compatibility symlinks against whichever base actually resolved, so the code stops rewiring `hf/` into a hard mount.
4. Report an unreachable candidate distinctly from an absent one, in the log and to `/api/health`.

## Related

- place#241 — same defect class, query path. Introduced `gateway/bounded_io.py`, which this should reuse.
- place#118 — the `/ix1` hard-mount single point of failure.

**Comment by docuracy** [comment]
Fixed and verified in production. `WorldHistoricalGazetteer/indexing@bb50f38`, deployed and running (gateway PID 2197105, listening ~5s after restart).

## Proof, against the actual wedged mount

The deployed resolver, run with the `SYMPHONYM_MODEL_DIR` pin removed so it takes the exact path that hung — in a throwaway process, so the serving gateway was untouched:

```
hf/final_model.pt -> /ix1/ishi/models/phonetic/checkpoints/v7/final_model.pt
hf/vocab          -> /ix1/ishi/models/phonetic/data/v7/vocab

RESOLVED in 4.01s -> /vast/ishi/elastic/hf
unreachable (skipped, not waited on):
    /vast/ishi/elastic/hf/final_model.pt
    /vast/ishi/elastic/hf/vocab/char_vocab.json
    /ix1/ishi/models/phonetic/checkpoints/v7/final_model.pt
   guard symphonym probe [repo-hf]: calls=4 timeouts=1 degraded=True
   guard symphonym probe [ix3]:     calls=1 timeouts=0 degraded=False
   guard symphonym probe [ix1]:     calls=1 timeouts=1 degraded=True
```

**4.01 seconds where it previously never returned.** It then falls through to `SymphonymModel`, which raises, which `app.lifespan` logs — so the gateway comes up without phonetic search instead of not coming up at all.

The `ix3` line is the other half: `timeouts=0, degraded=False`. The healthy candidate was still probed *after* two others had wedged.

## What changed

- Every probe, and the model load, runs behind `gateway.bounded_io.IoGuard` (place#241). An unresponsive candidate is skipped, never fatal, never blocking. The load raises `TimeoutError` on expiry rather than hanging.
- **One guard per candidate, not one shared.** I wrote it shared first, and the test passed — only because it set `cooldown=0`. A guard's breaker writes its resource off after a timeout, so a shared guard lets a wedged `/ix1` probe suppress the probes of the healthy `/vast` candidate behind it, and resolution fails with the model sitting on flash. The test now runs a 30s cooldown, which is what makes the separation observable instead of assumed; collapsing the guards fails exactly that one test and no other.
- `IX3_BASE` is tried before `IX1_BASE`, and the compatibility symlinks are planted against whichever base actually resolved — so the code no longer rewires its own fast path onto the hard mount.
- `unreachable` is now distinct from `absent` in the log and in a new `stores` block on `/api/health`, which also carries the place#241 hard-link guards:

```json
"stores": {
  "symphonym": {"loaded": true, "unreachable_paths": [], "probes": [...]},
  "hard_links": {"degraded": false, "stores": [
      {"label": "hard-link batch overlay", "calls": 4, "timeouts": 0, "degraded": false},
      {"label": "hard-link live-delta",    "calls": 4, "timeouts": 0, "degraded": false}]}
}
```

A wedged gateway could not report anything at all — it was blocked in a syscall rather than running code — so this is the state that was previously unobservable from outside.

## Controls

- **Negative control**: against the unfixed resolver, 4 of 7 tests fail and the run takes **210s instead of 3s** — the hang itself.
- **Mutation control**: collapsing the per-candidate guards onto one fails exactly the test written for that, and nothing else.
- Full suite: 1019 tests, same 6 pre-existing failures as HEAD (`restart_tileserver`, `aat_enrich`, `update_merge` — all untouched by this).

## Left in place deliberately

The two `hf/` symlinks into `/ix1` still exist. Deleting them is **not** sufficient on its own: `/vast` carries `phase3_best.pt`, not `final_model.pt`, so the CRC branch would find nothing there and replant the `/ix1` links as soon as that mount recovers. The durable state is the self-contained `/vast` copy pinned in `.env`:

```
SYMPHONYM_MODEL_DIR="/vast/ishi/models/phonetic/symphonym-v7-hf"
```

which now costs 4 seconds to fall back from rather than the service.

Related: place#241 (same defect class, query path — contributed `gateway/bounded_io.py`), place#118.

**Comment by docuracy** [comment]
**Correction to the "Left in place deliberately" section above — it is now wrong, and the `/ix1` symlinks are gone.**

That section said deleting the `hf/` symlinks would not be sufficient because "`/vast` carries `phase3_best.pt`, not `final_model.pt`". True at the time, and it framed the difference as an obstacle rather than as a question worth measuring. SG asked whether renaming the checkpoint would settle it permanently. It does.

## `phase3_best.pt` *is* the shipped model — measured, not assumed

Compared against the `model.safetensors` already verified to be serving correctly:

```
phase3_best.pt top-level keys: [model_state_dict, optimizer_state_dict, epoch, step, best_loss, config]
  epoch: 30
weight tensors in .pt: 42          weight tensors in safetensors: 42
keys only in .pt: (0)              keys only in safetensors: (0)

identical tensors: 42   differing: 0
```

All 42 bitwise identical. The 99 MB vs 33 MB difference is the optimizer state a training checkpoint carries and an inference artefact does not — not a different model.

## What was done

```bash
# zero bytes: the filesystem now guarantees the identity measured above
ln /vast/ishi/models/phonetic/checkpoints/v7/phase3_best.pt \
   /vast/ishi/models/phonetic/checkpoints/v7/final_model.pt

# unlink only — does not traverse, so it works with /ix1 still wedged
rm /vast/ishi/elastic/hf/final_model.pt /vast/ishi/elastic/hf/vocab
```

Both `hf/` entries are gitignored runtime artefacts (`hf/.gitignore:7,8`), so git neither tracks nor restores them.

## Result — `/ix1` is no longer in the path at all

Default resolution, pin removed, run in a throwaway process:

```
before: hf/final_model.pt: absent          before: hf/vocab: absent
RESOLVED in 0.03s -> /vast/ishi/elastic/hf
after:  hf/final_model.pt: symlink -> /vast/ishi/models/phonetic/checkpoints/v7/final_model.pt
after:  hf/vocab:          symlink -> /vast/ishi/models/phonetic/data/v7/vocab
unreachable: (none — nothing hung)
   guard symphonym probe [repo-hf]: calls=6 timeouts=0
   guard symphonym probe [ix3]:     calls=2 timeouts=0
MODEL LOADED in 2.11s
```

**0.03s, and no `ix1` guard was ever created** — the `/ix1` candidate is not reached, not merely survived. Compare the same run before this change: 4.01s, three unreachable paths, `ix1` guard degraded.

The re-planted symlinks point at `/vast`, which is the fix in this issue working as intended: symlinks are planted against whichever base actually resolved, and `IX3_BASE` is tried first.

## The two routes are interchangeable

`.pt` via the symlinks vs `safetensors` via the pin, quantised int8 output compared elementwise:

```
'Canterbury' en  identical=True     '北京' zh  identical=True
'Москва'     ru  identical=True     'القاهرة' ar  identical=True
'Reykjavík'  is  identical=True
```

So the default path and the pinned path reach the same model by different files. `SYMPHONYM_MODEL_DIR` stays in `.env` as an explicit belt to that braces, with its comment updated to say it is no longer load-bearing — the `.pt` route also has the advantage of needing only `torch`, where the safetensors route needs `safetensors` (declared in `hf/requirements.txt`, which the top-level `requirements.txt` does not pull in).

Standing caveat on the hardlink: the two names are one inode, so anything that rewrites `phase3_best.pt` *in place* changes `final_model.pt` with it. A new training run writing a new file is unaffected.
